host/sgx/sgxsign.c

```diff
@@ -664,3 +737,95 @@ oe_result_t oe_sgx_sign_enclave(
 
     return result;
 }
+
+oe_result_t oe_sgx_get_sigstruct_digest(
+    const OE_SHA256* mrenclave,
+    uint64_t attributes,
+    uint16_t product_id,
+    uint16_t security_version,
+    OE_SHA256* digest)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    sgx_sigstruct_t sigstruct;
+
+    if (digest)
+        memset(digest, 0, sizeof(OE_SHA256));
+
+    /* Check parameters */
+    if (!mrenclave || !digest)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* Initialize & sign the sigstruct */
+    OE_CHECK(_init_sigstruct(
+        mrenclave, attributes, product_id, security_version, &sigstruct));
+    OE_CHECK(_hash_sigstruct(&sigstruct, digest));
+
+    result = OE_OK;
+
+done:
+    return result;
+}
+
+oe_result_t oe_sgx_digest_sign_enclave(
+    const OE_SHA256* mrenclave,
+    uint64_t attributes,
+    uint16_t product_id,
+    uint16_t security_version,
+    const uint8_t* cert_pem_data,
+    size_t cert_pem_size,
+    const uint8_t* digest_signature,
+    size_t digest_signature_size,
+    sgx_sigstruct_t* sigstruct)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_cert_t cert = {0};
+    bool cert_initialized = false;
+    oe_rsa_public_key_t rsa = {0};
+    bool rsa_initalized = false;
+    OE_SHA256 digest = {0};
+
+    if (sigstruct)
```

Comment:
Consider keep this check and remove the same check below (`!sigstruct)` or remove this check and put the `memset`  after the checks of parameters.

Reply:
Same as above.